cmd/git-init/main.go

```diff
@@ -33,75 +28,12 @@ var (
 	path     = flag.String("path", "", "Path of directory under which git repository will be copied")
 )
 
-func run(logger *zap.SugaredLogger, cmd string, args ...string) error {
-	c := exec.Command(cmd, args...)
-	var output bytes.Buffer
-	c.Stderr = &output
-	c.Stdout = &output
-	if err := c.Run(); err != nil {
-		logger.Errorf("Error running %v %v: %v\n%v", cmd, args, err, output.String())
-		return err
-	}
-	return nil
-}
-
-func runOrFail(logger *zap.SugaredLogger, cmd string, args ...string) {
-	c := exec.Command(cmd, args...)
-	var output bytes.Buffer
-	c.Stderr = &output
-	c.Stdout = &output
-
-	if err := c.Run(); err != nil {
-		logger.Fatalf("Unexpected error running %v %v: %v\n%v", cmd, args, err, output.String())
-	}
-}
-
 func main() {
 	flag.Parse()
 	logger, _ := logging.NewLogger("", "git-init")
 	defer logger.Sync()
 
-	// HACK: This is to get git+ssh to work since ssh doesn't respect the HOME
-	// env variable.
-	homepath, err := homedir.Dir()
-	if err != nil {
-		logger.Fatalf("Unexpected error: getting the user home directory: %v", err)
+	if err := git.Fetch(*revision, *path, *url); err != nil {
```

Comment:
what do you think about this function taking `logger` as an argument? esp. since there's a logger created a few lines up

Reply:
Done!